src/details/ArborX_DetailsHappyTreeFriends.hpp

```diff
@@ -49,6 +51,51 @@ struct HappyTreeFriends
     return bvh.getBoundingVolume(node);
   }
 
+  template <class BVH>
+  static KOKKOS_FUNCTION auto const &getBoundingVolume(BVH const &bvh, int i)
+  {
+    return bvh._internal_and_leaf_nodes(i).bounding_volume;
+  }
+
+  template <class BVH>
+  static KOKKOS_FUNCTION bool isLeaf(BVH const &bvh, int i)
+  {
+    return bvh._internal_and_leaf_nodes(i).isLeaf();
+  }
+
+  template <class BVH>
+  static KOKKOS_FUNCTION auto getLeafPermutationIndex(BVH const &bvh, int i)
+  {
+    return bvh._internal_and_leaf_nodes(i).getLeafPermutationIndex();
+  }
+
+  template <class BVH>
+  static KOKKOS_FUNCTION auto getLeftChild(BVH const &bvh, int i)
+  {
+    return bvh._internal_and_leaf_nodes(i).left_child;
+  }
+
+  template <class BVH, std::enable_if_t<is_node_with_two_children<
+                           typename BVH::node_type>::value> * = nullptr>
+  static KOKKOS_FUNCTION auto getRightChild(BVH const &bvh, int i)
+  {
+    return bvh._internal_and_leaf_nodes(i).right_child;
+  }
+
+  template <class BVH, std::enable_if_t<is_node_with_left_child_and_rope<
+                           typename BVH::node_type>::value> * = nullptr>
+  static KOKKOS_FUNCTION auto getRightChild(BVH const &bvh, int i)
+  {
+    return bvh._internal_and_leaf_nodes(getLeftChild(bvh, i)).rope;
+  }
+
+  template <class BVH, std::enable_if_t<is_node_with_left_child_and_rope<
+                           typename BVH::node_type>::value> * = nullptr>
+  static KOKKOS_FUNCTION auto getRope(BVH const &bvh, int i)
+  {
+    return bvh._internal_and_leaf_nodes(i).rope;
+  }
```

Comment:
It would be really weird if these functions were not all inlined.

Reply:
What are you saying?

Reply:
I was just commenting on the weirdness of the performance improvements possibly related to small functions like these.